### DISCUSSION TITLE
Rechenoperation in Script fehlerhaft

### DISCUSSION THREAD

**Discussion by Weseler485** [OpenCCU/OpenCCU#2389] [🇩🇪 Fragen&Antworten] [answered]
### Describe the issue you are experiencing

Das Script rechnet falsch, wenn man zwei Rechenoperationen hintereinander ausführt.

### Describe the behavior you expected

Was ich erwarte...

### Steps to reproduce the issue

real r_failure = 1.23;
r_failure = r_failure + 1 - 0.5;
WriteLine(r_failure);

Ausgabe: 2.23

### What is the version this bug report is based on?

3.69.7.20230626

### Which base platform are you running?

rpi3 (RaspberryPi3)

### Which HomeMatic/homematicIP radio module are you using?

RPI-RF-MOD

### Anything in the logs that might be useful for us?

```txt
...
```

**Comment by MichaelN0815** [comment]
Das ist vollkommen in Übereinstimmung mit der Dokumentation.

**Comment by jens-maus** [comment, marked as answer]
Genau so ist es, denn entgegen der eigentlich intuitiven Erwartung rechnet die ReGaHss bei diesem Beispiel wie folgt:

1. `1 - 0.5 = 1` weil hier die `1` auf der linken seite dafür sorgt das die `0.5` auf der rechten seite auch in ein integer gewandelt werden vor der Subtraktion,, ergo `0`. Und somit kommt dabei dann 1 als integer raus.
2. Als nächstes wird dann `r_failure + 1` gerechnet und somit die 1 (integer) aus der vorherigen Berechnung in ein real/float gewandelt und dann zu `r_failure` addiert.

Folglich ist also das Ergebnis von `2.26` vollkomen in Ordnung - einfach weil hier 1. von rechts nach links gerechnet wird und 2. der konkrete Datentyp der einzelnen Werte relevant ist.

Möchte man also das eigentlich zu erwartende Ergebnis von 1.73 erhalten muss man  die Berechnung wie folgt vornehmen:

```
r_failure = r_failure + 1.0 - 0.5;
```

d.h. also statt der `1` (integer) explizit ein real/float von `1.0` schreiben.

**Reply by Weseler485** [reply]
Super, danke für den Workaround.
Zu simpel um zu später Stunde selbst drauf zu kommen.

**Reply by jens-maus** [reply]
Das ist kein "Workaround": "it‘s a feature, not a bug" 😜
